### DISCUSSION TITLE
stable version 0.7.4 -rc 0.7.3-r52 wlroots 0.17 0.18

### DISCUSSION THREAD

**Discussion by fungilife** [labwc/labwc#2038] [Q&A] [answered]
Am I missing something with version order?
When wlroots0.18 came out 0.7.3 would only build with 0.17 but a few rc versions later it would build with wlroots 0.18 and has worked reasonably well.
Building every rc version since and seeing 0.7.3 still being the latest I didn't even notice the 0.7.4 release. which still does not build with wlroots 0.18

Is this an intentional order of a stable release build on 0.17 and no 0.7.4-rc ?

**Comment by Consolatis** [comment, marked as answer]
That is just because the master branch doesn't yet carry a new version (should be something like `0.8-dev`).
The v0.7 branch is in maintenance mode and will keep using wlroots 0.17 whereas the master branch will follow wlroots 0.18 but there wasn't a release yet (and there won't be one for the next 1 to 2 months).

All fixes applied to the v0.7 branch (and thus ending up in releases like 0.7.4) are also applied on the master branch.

**Comment by johanmalm** [comment]
We ought to consider making this clearer in NEWS.md

**Reply by fungilife** [reply]
I think if I understand the process correctly that the next commit on the master branch should be given a tag as 0.7.4-r1 instead of 0,7,3-rc66

**Reply by Consolatis** [reply]
The version shown by `labwc --version` is generated automatically by git. As 0.7.3 is the latest release from the master branch it will keep being `0.7.3-$N-g$C` with `$N` being the number of commits since 0.7.3 and `$C` the current commit hash. This will change only once we release 0.8.0. For more see
- #2032

In general: all 0.7.x releases of labwc require wlroots 0.17.x and all 0.8.x releases of labwc require wlroots 0.18.x (same story for 0.5 and 0.6).
